src/main/java/com/couchbase/lite/replicator/BulkDownloader.java

```diff
@@ -134,47 +136,45 @@ protected void executeRequest(HttpClient httpClient, HttpUriRequest request) {
                         InputStream inputStream = null;
                         try {
                             inputStream = entity.getContent();
-
-                            Header contentTypeHeader = entity.getContentType();
-                            if (contentTypeHeader != null) {
-                                // multipart
-                                if (contentTypeHeader.getValue().contains("multipart/")) {
-                                    Log.v(Log.TAG_SYNC, "contentTypeHeader = %s",
-                                            contentTypeHeader.getValue());
-                                    _topReader = new MultipartReader(contentTypeHeader.getValue(), this);
-                                    byte[] buffer = new byte[BUF_LEN];
-                                    int numBytesRead = 0;
-                                    while ((numBytesRead = inputStream.read(buffer)) != -1) {
-                                        _topReader.appendData(buffer, 0, numBytesRead);
+                            GZIPInputStream gzipStream = null;
+                            try {
+                                Header contentTypeHeader = entity.getContentType();
+                                if (contentTypeHeader != null) {
+                                    // decompress if contentEncoding is gzip
```

Comment:
If contentTypeHeader is null, the inputStream created at line 138 doesn't get used. Can we move `inputStream = entity.getContent();` inside the `if (contentTypeHeader != nulll)` block?